Mathlib/CategoryTheory/Discrete/Basic.lean

```diff
@@ -172,6 +176,14 @@ theorem functor_map {I : Type u₁} (F : I → C) {i : Discrete I} (f : i ⟶ i)
 theorem functor_obj_eq_as {I : Type u₁} (F : I → C) (X : Discrete I) :
     (Discrete.functor F).obj X = F X.as :=
   rfl
+
+@[ext]
+lemma functor_ext {I : Type u₁} {G F : Discrete I ⥤ C} (h : (i : I) → G.obj ⟨i⟩ = F.obj ⟨i⟩) :
+    G = F := by
+  fapply Functor.ext
+  · intro I; rw [h]
+  · intro ⟨X⟩ ⟨Y⟩ ⟨⟨p⟩⟩; simp at p; induction p; simp
```

Comment:
This is a nonterminal simp: it should be squeezed (by replacing `simp at p` by the output of `simp?`).